### PR TITLE
magento2/issues/29479 Returns : graphql implementation

### PR DESCRIPTION
**Pull request by sasha19957099**
## Problem

Working with ticket https://github.com/magento/magento2/issues/29479 I see that in RMA module we have additional fields in the request (order_id, customer_custom_email)

### CODE REVIEW

File: design-documents/graph-ql/coverage/returns.graphqls [hunk 1]
```diff
@@ -13,6 +13,8 @@ type Mutation {
 }
 
 input RequestReturnInput {
+    order_id: ID!
+    customer_custom_email: String!
```
Comment (melnikovi) [inline design-documents/graph-ql/coverage/returns.graphqls:17]:
```suggestion
    contact_email: String
```

Let's make field optional as it's optional in the Luma.

**Comment by nrkapoor** [comment]
@melnikovi can we also update all the "id" fields to "uid" to be consistent with the new proposal by @DrewML?

File: design-documents/graph-ql/coverage/returns.md [hunk 2]
```diff
@@ -211,6 +211,8 @@ Existing schema of `Attribute` and `AttributeOption` must be extended to provide
 mutation {
   requestReturn(
     input: {
+      order_id: 12345
```
Comment (melnikovi) [inline design-documents/graph-ql/coverage/returns.md:214]:
```suggestion
      order_id: 000000001
```

File: design-documents/graph-ql/coverage/returns.graphqls [hunk 3]
```diff
@@ -13,6 +13,8 @@ type Mutation {
 }
 
 input RequestReturnInput {
+    order_id: ID!
```
Comment (melnikovi) [inline design-documents/graph-ql/coverage/returns.graphqls:16]:
```suggestion
    order_uid: ID!
```

Suggestion from @nrkapoor, pending discussion.

File: design-documents/graph-ql/coverage/returns.md [hunk 4]
```diff
@@ -211,6 +211,8 @@ Existing schema of `Attribute` and `AttributeOption` must be extended to provide
 mutation {
   requestReturn(
     input: {
+      order_id: 000000001
```
Comment (melnikovi) [inline design-documents/graph-ql/coverage/returns.md:214]:
```suggestion
      order_uid: MQ==
```

Suggestion from @nrkapoor, pending discussion.

**Comment by melnikovi** [comment]
Currently `CustomerOrder` type has `id` and `number` fields, in the future id will be replaced with `uid`.

- `id` - unique identifier for the order
- `number` - order number that is visible to a customer; not uniquer for all orders; unique in the scope of website and store; it's also sequential; generation can be customized
- `uid` - see [this](https://github.com/magento/architecture/pull/415) proposal for mode details

We can use `uid` and `number` fields in the mutations input types that reference CustomerOrder`, here are some pros and cons.

`uid` - will be aligned with other places, we are going to a state when types reference each other by `uid` and `uid` used in mutations.
`number` - in some cases it would make more sense for querying data. If we suggest to use it for mutations it might not follow the pattern used in other mutations and be confusing for developers.

We can't have just `uid` and store value of `number` there as `uid` supposed to be unique and `number` currently is not unique. Not having `uid` and having `number` for `CustomerOrder` has similar implications.

It makes sense to have both fields and use `uid` in mutations that reference `CustomerOrder` type.

**Comment by paliarush** [comment]
`uid` should be of type `ID` (not human-readable)
`number` should be of type `String` (human-readable)

File: design-documents/graph-ql/coverage/returns.graphqls (hunk 3 above)
Comment (melnikovi) [reply design-documents/graph-ql/coverage/returns.graphqls:16]: Discussed with @nrkapoor and agree to replace `id` with `uid` everywhere in schema for returns.

**Pull request merged**